### PR TITLE
Bug fix/rotate

### LINKED ISSUE #3: Asteroid Belt is not rotating in VR
The rotating asteroid belt only seems to be working in the 3D scene, but not in the vr scene. In general the scene decoder iterates over the asteroids and apllies the script, but the _process(delta) is never called on them in vr.

### LINKED ISSUE #4: Torus is a square
The torus logic, which is for example used in the ring of saturn is currently a square, not a torus, see image below.

grafik

### PR DESCRIPTION
**Pull request by kevinmatthes**
This Pull Request aims to fix #3 and #4.

### CODE REVIEW

File: project/scenes/scene_decoder.gd [hunk 1]
```diff
@@ -261,11 +261,8 @@ func create_object(object: Dictionary, geometries: Dictionary, materials: Dictio
 	elif "centre" in object && "radius" in object && (!object.has("with_script") || !object.with_script):
 		var centre := get_randomized_vector3(object.centre, rng)
 		var radius: float = get_randomized(object.radius, rng)
-		node.transform.origin = centre + radius * Vector3(
-			cos(rng.randf_range(0, 2 * PI)),
-			sin(rng.randf_range(0, 2 * PI)),
-			0
-		)
+		var phi := rng.randf_range(0, 2 * PI)
+		node.transform.origin = centre + radius * Vector3(cos(phi), sin(phi), 0)
```
Comment (kevinmatthes) [inline project/scenes/scene_decoder.gd:265]: This looks basically like a scaled and translated unit circle -- a *full* unit circle.

Is there an area around the centre cut out or does this really mean that the circle intersects the planet surrounded by the belt, meaning some objects *might* be created within it?

**Comment by kevinmatthes** [comment]
I drafted this Pull Request since there might be a performance leak due to the definition of the coordinate area.

File: project/scenes/scene_decoder.gd (hunk 1 above)
Comment (kevinmatthes) [reply project/scenes/scene_decoder.gd:265]: Okay, now I understood.  It is a cylinder with a cut-out centre.

**Review by NiklasLeukroth — CHANGES_REQUESTED** [review]
This commit should not be included, since the changes will be obsolete after the fix. They therefore do not belong in the merge.

**Comment by kevinmatthes** [comment]
The changes can just be reverted before the merge.  All commits will be squashed anyway, so that does not matter.  Alternative:  rebase the commit into a new branch and force push the other commits after it.

**Pull request closed without merging**